src/OrleansAWSUtils/Membership/DynamoDBMembershipTable.cs

```diff
@@ -0,0 +1,347 @@
+
+                MembershipTableData data = Convert(records);
+                if (logger.IsVerbose2) logger.Verbose2("ReadAll Table=" + Environment.NewLine + "{0}", data.ToString());
+
+                return data;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error reading all silo entries {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<bool> InsertRow(MembershipEntry entry, TableVersion tableVersion)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("InsertRow entry = {0}", entry.ToFullString());
+                var tableEntry = Convert(entry);
+
+                bool result;
+
+                try
+                {
+                    var expression = $"attribute_not_exists({SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME}) AND attribute_not_exists({SiloInstanceRecord.SILO_IDENTITY_PROPERTY_NAME})";
+
+                    await storage.PutEntryAsync(TABLE_NAME_DEFAULT_VALUE, tableEntry.GetFields(true), expression);
+                    result = true;
+                }
+                catch (ConditionalCheckFailedException)
+                {
+                    result = false;
+                    logger.Warn(ErrorCode.MembershipBase,
+                        $"Insert failed due to contention on the table. Will retry. Entry {entry.ToFullString()}");
+                }
+                    
+                return result;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error inserting entry {entry.ToFullString()} to the table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<bool> UpdateRow(MembershipEntry entry, string etag, TableVersion tableVersion)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("UpdateRow entry = {0}, etag = {1}", entry.ToFullString(), etag);
+                var siloEntry = Convert(entry);
+                int currentEtag = 0;
+                if (!int.TryParse(etag, out currentEtag))
+                {
+                    logger.Warn(ErrorCode.MembershipBase,
+                        $"Update failed. Invalid ETag value. Will retry. Entry {entry.ToFullString()}, eTag {etag}");
+                    return false;
+                }
+
+                siloEntry.ETag = currentEtag + 1;
+
+                bool result;
+
+                try
+                {
+                    var conditionalValues = new Dictionary<string, AttributeValue> { { CURRENT_ETAG_ALIAS, new AttributeValue { N = etag } } };
+                    var etagConditionalExpression = $"{SiloInstanceRecord.ETAG_PROPERTY_NAME} = {CURRENT_ETAG_ALIAS}";
```

Comment:
Rethinking on that, no... I think we shouldn't change the filter and leave only the etag. Since the ETag is never null (initial value is 0), and the row will never be inserted without DEPLOYMENT_ID and SILO_IDENTITY since they are the keys, there is no point in a row to has ETag but no keys :)